Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/presenter/ImageViewerPresenter.java

```diff
@@ -83,14 +106,27 @@ public ImageViewerPresenter(Callback callback) {
                 || !audioManager.isWiredHeadsetOn());
     }
 
+    @SuppressLint("UseSparseArrays")
     public void showImages(List<PostImage> images, int position, Loadable loadable) {
         this.images = images;
-        selectedPosition = Math.max(0, Math.min(images.size() - 1, position));
         this.loadable = loadable;
+        this.selectedPosition = Math.max(0, Math.min(images.size() - 1, position));
+        this.progress = new HashMap<>(images.size());
 
-        progress = new ArrayList<>(images.size());
-        for (int i = 0; i < images.size(); i++) {
-            progress.add(i, -1f);
+        int chunksCount = ChanSettings.ConcurrentFileDownloadingChunks.toChunkCount(
```

Comment:
I did it via ListSettingView because I didn't want the user to be able to enter the number by himself. I rewrote it a little bit so now it looks like similar setting - backgroundTimeout.

Reply:
Is there a reason you don't want the user to enter a chunk amount by themselves? You can add in some code to IntegerSettingView to limit the range of input values.

Reply:
After testing it for couple of week I can say that there is no point in splitting a file in more than 4 chunks. So 1, 2 and 4 should be enough. There is just no benefit after 4 chunks.

Reply:
You didn't answer my question. Why would you not allow 3 chunks?

Reply:
Because it's not a power of two.

Reply:
I mean, what's the point of having the user input a value if there only three (well maybe four) options?

Reply:
Code reuse? And that still doesn't answer my question; "it's not a power of two" doesn't explain WHY it needs to be a power of two. If I'm downloading a 24KB file with a minimum chunk size of 8KB, then 1, 2, 3, 4, 6, 8, etc. chunks are all divisible without remainder.

Reply:
There is no specific reason for why I didn't include 3 into the options other than it not being a power of two. I can include it if you want.

>If I'm downloading a 24KB file with a minimum chunk size of 8KB, then 1, 2, 3, 4, 6, 8, etc. chunks are all divisible without remainder.

It really doesn't matter whether it's 2 or 3 or 4 other than a little bit of speed up with a higher number (which gets to the point of no difference after the 4). Even if I use a prime number for the chunks count it still will be fine, because if the total amount of bytes is not divisble by the chunk count without the remainder then the remainer will be added up to the last chunk.

Reply:
Another reason for using a predifined list of options instead of allowing the user to enter a number by himself is that the user does not know what range he can enter so he will probably try entering something like 100, then he will see an error message, then 50, error message again and so on until he enters a number between 1-4. And then the user may ask a question: "Why in the fuck was it not implemented with a predifined list of options if there are only 4 possible options?"